### PR TITLE
Remove Hive integration tests for CDH 5

### PR DESCRIPTION
**Pull request by electrum**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

refactoring

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Review by dain — APPROVED** [review]
Nice!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -174,9 +174,7 @@ jobs:
       fail-fast: false
       matrix:
         config:
-          - config-empty
```
Comment (hashhar) [inline .github/workflows/ci.yml:177]: Why did you remove config-empty? it's used to test metastore only operations AFAIK. cc: @findepi

Comment (findepi) [inline .github/workflows/ci.yml:177]: This probably wasn't CDH5

File: plugin/trino-hive-hadoop2/conf/hive-tests-defaults.sh [hunk 2]
```diff
@@ -9,4 +9,3 @@ then
 fi
 
 export DOCKER_IMAGES_VERSION=${DOCKER_IMAGES_VERSION:-$DEFAULT_DOCKER_VERSION}
-export HADOOP_BASE_IMAGE="${HADOOP_BASE_IMAGE:-ghcr.io/trinodb/testing/hdp2.6-hive}"
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/conf/hive-tests-defaults.sh:12]: defaults should still declare the HADOOP_BASE_IMAGE

File: .github/workflows/ci.yml (hunk 1 above)
Comment (electrum) [reply .github/workflows/ci.yml:177]: You're right, this should have gone in the other commit to remove pre Hive 3.x

File: plugin/trino-hive-hadoop2/conf/hive-tests-defaults.sh (hunk 2 above)
Comment (electrum) [reply plugin/trino-hive-hadoop2/conf/hive-tests-defaults.sh:12]: Why? We always run with a specific version. The concept of "default" seems confusing -- explicit is better.

Comment (findepi) [reply plugin/trino-hive-hadoop2/conf/hive-tests-defaults.sh:12]: The concept of "default" has it's usage -- you want to allow simple invocation of scrips from command line, for developer's consumption, without having to specify the config (both in hive tests and in product tests)
otherwise devs need to source config files, whicih is error-prone in a (long running) interactive shell
